maze-consulting/SFCC_Klaviyo_Maze
# Feature/kscu 57 kl client side debugging

## Description

This update handles an ability to activate 'Debugging Mode' by entering `kldebug=true` as a URL query. With this PR, debugging mode is available for multiple Klaviyo Events in both SFRA & SiteGen sites:

- 'Viewed Product'
- 'Viewed Category'
- 'Searched Site'
- 'Started Checkout'
- 'Added to Cart'

Once debug mode has been activated using `kldebug=true`, Debug Data will be displayed in the console. This update also includes a fix for a compatibility mode issue that was creating a faulty cart recreate link in 21.7 compatibility mode.

## Manual Testing Steps

This was tested in both SFRA & SiteGen across multiple events. Manual tests included:

- 'Viewed Product' _(Logged in / Logged out)_
- 'Viewed Category' _(Logged in / Logged out)_
- 'Searched Site' _(Logged in / Logged out)_
- 'Started Checkout' _(Logged in / Logged out)_
- 'Added to Cart' _(Logged in / Logged out)_

As mentioned in the ticket, the 'Order Confirmation' event has been excluded from this checkout and is no longer a part of [KSCU-57](https://themazegroup.atlassian.net/browse/KSCU-57). 


## Reviewer Note:

This update contains adjustments to two SiteGen controllers: _Cart.js & COCustomer.js._ The updates from these controllers should be pasted into the original files within your Site Genesis project structure in order for this branch to work locally. 


## BugFix Note:

During development of this branch, a discrepancy was discovered with the cart rebuilding link that was created in environments with 21.7 compatibility mode. The adjustment in startedCheckout JS ([this commit](https://github.com/maze-consulting/SFCC_Klaviyo_Maze/pull/21/commits/8d35753fede0618b793b02c817d9389c19238174)) fixes the 21.7 compatability mode issue and clears the path for cart rebuilding links to be successfully built in 22.7 AND 21.7.

## Pre-Submission Checklist:

[ x ] This update successfully builds
[ x ] No console errors are displayed on the screen with these updates
[ x ] Console Logs & debugging created during development for debugging that are no longer needed have been removed
[ n/a ] ToDo comments were left as necessary for future reference
[ x ] A description of this update & references to the manual tests has been included in this PR
